airflow-core/src/airflow/secrets/local_filesystem.py

```diff
@@ -233,10 +233,23 @@ def load_variables(file_path: str) -> dict[str, str]:
     log.debug("Loading variables from a text file")
 
     secrets = _parse_secret_file(file_path)
-    invalid_keys = [key for key, values in secrets.items() if isinstance(values, list) and len(values) != 1]
-    if invalid_keys:
-        raise AirflowException(f'The "{file_path}" file contains multiple values for keys: {invalid_keys}')
-    variables = {key: values[0] if isinstance(values, list) else values for key, values in secrets.items()}
+
+    # Determine a file format to handle lists correctly
+    ext = file_path.rsplit(".", 2)[-1].lower()
```

Comment:
I'd rather use `Pathlib` method to determine file's suffix rather than string manipulation